src/Promise.php

```diff
@@ -41,7 +41,7 @@ interface Promise
      *
      * @return Promise a new resolved promise with value of the executed callback (onFulfilled / onRejected)
      */
-    public function then(callable $onFulfilled = null, callable $onRejected = null);
+    public function then(?callable $onFulfilled = null, ?callable $onRejected = null);
```

Comment:
afaik this is a BC break for anything implementing the promise, right?

if it is, we should bump the major version number.

Reply:
No, the code is 100% equivalent. This has always been a nullable type.

Reply:
Here's an example where I examine the type with reflection:

https://3v4l.org/EQimk

One parameter is declared `callable`, the other `?callable`. As you can see in the output, both parameters look the same.

Reply:
thanks. i also added a class that implements the interface and changes `?`
at least up to 8.3 this is indeed equivalent.

thanks for the clarification, then we can savely merge this.